src/Streamly/Coreutils/Rm.hs

```diff
@@ -16,32 +16,37 @@ module Streamly.Coreutils.Rm
     )
 where
 
-import Data.Default.Class (Default(..))
-import GHC.Generics (Generic)
 import Streamly.Coreutils.Common (Switch(..))
+import Streamly.Coreutils.FileTest (isFile, test)
 import System.Directory
     ( removeFile
     , removeDirectoryRecursive
     , removePathForcibly
     )
+import Control.Monad (when)
 
 data Rm = Rm
     {
       force :: Switch
     , recursive :: Switch
-    } deriving (Generic, Eq , Show)
+    }
 
-instance Default Rm
+defaultConfig :: Rm
+defaultConfig = Rm Off Off
 
 rmForce :: Switch -> Rm -> Rm
-rmForce sw options = options {force = sw}
+rmForce sw cfg = cfg {force = sw}
 
 rmRecursive :: Switch -> Rm -> Rm
-rmRecursive sw options = options {recursive = sw}
+rmRecursive sw cfg = cfg {recursive = sw}
 
-rm :: Rm -> FilePath -> IO ()
-rm opt = do
+rm :: (Rm -> Rm) -> FilePath -> IO ()
+rm f path = do
+    let opt = f defaultConfig
     case opt of
-        Rm {force = On} -> removePathForcibly
-        Rm {recursive = On} -> removeDirectoryRecursive
-        _ -> removeFile
+        Rm {force = On, recursive = On} -> removePathForcibly path
+        Rm {force = On, recursive = Off} -> do
+            notDir <- test path isFile
+            when notDir $ removePathForcibly path
```

Comment:
Can not use removePathForcibly without the `nuke` options, as explained above.

Here we should do:

* Check if the file exists, if does not exist then do nothing
* If it exists then call removeFile